tinylfu/tinylfu.go

```diff
@@ -0,0 +1,122 @@
+// Package tinylfu is an implementation of the W-TinyLFU caching algorithm.
+// See details at http://arxiv.org/abs/1512.00727
+package tinylfu
+
+// Policy implements a windowed TinyLFU eviction policy. It is not safe for concurrent access.
+type Policy struct {
+	data     map[uint64]*element
+	admittor AdmissionPolicy
```

Comment:
we can potentially move the configuration to another struct and define all the functions (With*) on that config instead. Check this out https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis